src/server/api.php

```diff
@@ -2028,6 +2092,44 @@ function meta_delphi() {
       $epidata = get_covid_hosp_facility_lookup($state, $ccn, $city, $zip, $fips_code);
       store_result($data, $epidata);
     }
+  } else if($source === 'covidcast_nowcast') {
+    if(require_all($data, array('data_source', 'time_type', 'geo_type', 'time_values'))
+       && require_any($data, array('signal', 'signals'))
+       && require_any($data, array('geo_value', 'geo_values'))) {
+      // parse the request
+      $time_values = extract_dates($_REQUEST['time_values']);
+      $as_of = isset($_REQUEST['as_of']) ? parse_date($_REQUEST['as_of']) : null;
+      $issues = isset($_REQUEST['issues']) ? extract_dates($_REQUEST['issues']) : null;
+      $lag = isset($_REQUEST['lag']) ? intval($_REQUEST['lag']) : null;
+      $signals = extract_values(isset($_REQUEST['signals']) ? $_REQUEST['signals'] : $_REQUEST['signal'], 'string');
+      $geo_values = isset($_REQUEST['geo_value']) ? $_REQUEST['geo_value'] : extract_values($_REQUEST['geo_values'], 'string');
+      // get the data
+      $epidata = get_covidcast_nowcast(
+          $_REQUEST['data_source'],
+          $signals,
+          $_REQUEST['time_type'],
+          $_REQUEST['geo_type'],
+          $time_values,
+          $geo_values,
+          $as_of,
+          $issues,
+          $lag);
+      if(isset($_REQUEST['format']) && $_REQUEST['format']=="tree") {
+        //organize results by signal
+        $epi_tree = array();
+        $key = -1;
+        foreach ($epidata as $row) {
+          if ($key != $row['signal']) {
+            $key = $row['signal'];
+            $epi_tree[$key] = array();
+          }
+          unset($row['signal']);
+          array_push($epi_tree[$key],$row);
+        }
+        $epidata = array($epi_tree);
+      }
```

Comment:
recommend dropping this unless you have an actual use case for multi-signal queries with results grouped by signal

Reply:
good call, will remove.